packages/fx-core/src/core/collaborator.ts

```diff
@@ -124,6 +143,97 @@ export class CollaborationUtil {
       isAdministrator,
     };
   }
+
+  static async loadDotEnvFile(
+    dotEnvFilePath: string
+  ): Promise<Result<{ [key: string]: string }, FxError>> {
+    try {
+      const result: { [key: string]: string } = {};
+      if (!(await fs.pathExists(dotEnvFilePath))) {
+        throw new Error(getLocalizedString("core.collaboration.error.dotEnvFileNotExist"));
+      }
+
+      const envs = dotenv.parse(await fs.readFile(dotEnvFilePath));
+      const entries = Object.entries(envs);
+      for (const [key, value] of entries) {
+        result[key] = value;
+      }
+      return ok(result);
+    } catch (error: any) {
+      return err(
+        new UserError(
+          SolutionSource,
+          SolutionError.FailedToLoadDotEnvFile,
+          getLocalizedString("core.collaboration.error.failedToLoadDotEnvFile", error?.message)
+        )
+      );
+    }
+  }
+
+  // Priority parameter > dotenv > env
```

Comment:
From the implementation, seems priority is env > dotenv > parameter ?

Reply:
In the implementation, to support scenario that user may have one id in env and another in dotenv, we will continue try to read id if not all two ids are provided. If app id exists in parameter, value in env/dotenv will not replace the value.